core/payment/examples/debit_note_flow.rs

```diff
@@ -0,0 +1,113 @@
+use bigdecimal::BigDecimal;
+use chrono::Utc;
+use std::time::Duration;
+use ya_client::payment::{PaymentProviderApi, PaymentRequestorApi};
+use ya_client::web::WebClient;
+use ya_client_model::payment::{Acceptance, DocumentStatus, NewAllocation, NewDebitNote};
+
+#[actix_rt::main]
+async fn main() -> anyhow::Result<()> {
+    std::env::set_var("RUST_LOG", "info");
+    env_logger::init();
+
+    let client = WebClient::builder().build();
+    let provider: PaymentProviderApi = client.interface()?;
+    let requestor: PaymentRequestorApi = client.interface()?;
+
+    log::info!("Issuing debit note (total amount due: 1 GNT)...");
+    let debit_note = provider
+        .issue_debit_note(&NewDebitNote {
+            activity_id: "activity_id".to_string(),
+            total_amount_due: BigDecimal::from(1u64),
+            usage_counter_vector: None,
+            payment_due_date: Some(Utc::now()),
+        })
+        .await?;
+    log::info!("Debit note issued.");
+
+    log::info!("Sending debit note...");
+    provider.send_debit_note(&debit_note.debit_note_id).await?;
+    log::info!("Debit note sent.");
+
+    log::info!("Creating allocation...");
+    let allocation = requestor
+        .create_allocation(&NewAllocation {
+            total_amount: BigDecimal::from(10u64),
+            timeout: None,
+            make_deposit: false,
+        })
+        .await?;
+    log::info!("Allocation created.");
+
+    log::info!("Accepting debit note...");
+    let now = Utc::now();
+    requestor
+        .accept_debit_note(
+            &debit_note.debit_note_id,
+            &Acceptance {
+                total_amount_accepted: debit_note.total_amount_due.clone(),
+                allocation_id: allocation.allocation_id.clone(),
+            },
+        )
+        .await?;
+    log::info!("Debit note accepted.");
+
+    log::info!("Waiting for payment...");
+    let timeout = Some(Duration::from_secs(300)); // Should be enough for GNT transfer
```

Comment:
is there an issue to make a proper function for this?

Reply:
What do you mean by a proper function?

Reply:
My bad, mis-read this line for a sleep, ignore please!